src/services/osmApi.ts

```diff
@@ -94,10 +94,38 @@ const getCountryCode = async (feature: Feature): Promise<string | null> => {
   return null;
 };
 
+const getElementsAndCenter = async (
+  apiId: OsmId,
+): Promise<{ element: any; center: false | LonLat }> => {
+  const elementPromise = getOsmPromise(apiId);
+  switch (apiId.type) {
+    case 'node':
+      const elementNode = await elementPromise;
+      return { element: elementNode, center: false as const };
+    case 'way':
+      const [elementWay, centerWay] = await Promise.all([
+        elementPromise,
+        getCenterPromise(apiId),
+      ]);
+      return { element: elementWay, center: centerWay };
+    case 'relation':
+      const elementRel = await elementPromise;
+      const [firstMember] = elementRel.members;
```

Comment:
There is behaviour of getCenterPromise which this is missing. It handles the scenario when user clicks a relation in his browser, and the center is thus stored in the featureCenterCache. The marker is then displayed exactly where user clicked, which is better UX. Also skeleton would show the same marker, and we don' want it to jump.

Most importantly it skips the overpassCenter request in browser, and performance is our most valuable asset. Once we spoil it, it is much harder to get it back.

Please read https://github.com/zbycz/osmapp/wiki/ARCHITECTURE how it works.

Reply:
Also maybe change the behaviour only for route/RouteMasters first. I am not sure how to make this correctly, but for some relations (like https://openclimbing.org/relation/17262675) it places the marker in SSR mode on almost the same place as where the icon lays. Which is right. We should keep this, and maybe later we will find a way how to make a general approach to this.

Reply:
Ok, I now didn't implement the role=label behaviour as it would be unnecessary on routes and train masters.